src/main/java/org/ice4j/ice/ConnectivityCheckClient.java

```diff
@@ -972,6 +979,13 @@ protected void run()
          */
         protected Duration getDelayUntilNextRun()
```

Comment:
Is it better to do this in `schedule`?

Reply:
`schedule` is in the base class and there are other users of `PeriodicRunnable.`  We could have a `firstRunImmediate` flag in the base class which the subclass sets but it seemed unnecessarily complicated?